filelog.go

```diff
@@ -128,12 +128,21 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 			}
 		}()
 
+		// Failure counters
+		var rotationFailures uint64
+		var writeFailures uint64
+
 		for {
 			select {
 			case <-w.rot:
 				if err := w.handleRotate(time.Now()); err != nil {
 					fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.filename, err)
-					return
+					rotationFailures += 1
+				} else {
+					if rotationFailures != 0 {
+						_, logErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Ignored %d previous rotation attempts due to rotation failure\n", w.filename, rotationFailures)
+						if logErr == nil { rotationFailures = 0 }
```

Comment:
I can definitely see the `logErr` variable confusion - will fix that. 

What I think you're saying is to restructure the handling on `handleRotate` failure to only print one message on rotation failure, and only to print the number of failures if it failed more than once and we couldn't report the previous failure? That should be doable.